package com.google.android.gms.common.data;

/* compiled from: com.google.android.gms:play-services-base@@18.4.0 */
/* loaded from: classes.dex */
public interface DataBuffer<T> extends java.lang.Iterable<T>, com.google.android.gms.common.api.Releasable, java.io.Closeable {
    void close();

    T get(int r1);

    int getCount();

    @com.google.android.gms.common.annotation.KeepForSdk
    android.os.Bundle getMetadata();

    @java.lang.Deprecated
    boolean isClosed();

    @Override // java.lang.Iterable
    java.util.Iterator<T> iterator();

    void release();

    java.util.Iterator<T> singleRefIterator();
}
